lib/inspec/utils/run_data_filters.rb

```diff
@@ -0,0 +1,97 @@
+
+      # Long name, but we want to be clear this operates on the Hash
+      # This is the only method that client libraries need to call; any future
+      # feature growth should be handled internally here.
+      def apply_run_data_filters_to_hash
+        @config[:runtime_config] = Inspec::Config.cached || {}
+        apply_report_resize_options
+        redact_sensitive_inputs
+        suppress_diff_output
+        sort_controls
+      end
+
+      # Apply options such as message truncation and removal of backtraces
+      def apply_report_resize_options
+        runtime_config = @config[:runtime_config]
+
+        message_truncation = runtime_config[:reporter_message_truncation] || "ALL"
+        trunc = message_truncation == "ALL" ? -1 : message_truncation.to_i
+        include_backtrace = runtime_config[:reporter_backtrace_inclusion].nil? ? true : runtime_config[:reporter_backtrace_inclusion]
+
+        @run_data[:profiles]&.each do |p|
+          p[:controls].each do |c|
+            c[:results]&.map! do |r|
+              r.delete(:backtrace) unless include_backtrace
+              if r.key?(:message) && r[:message] != "" && trunc > -1
+                r[:message] = r[:message][0...trunc] + "[Truncated to #{trunc} characters]"
+              end
+              r
+            end
+          end
+        end
+      end
+
+      # Find any inputs with :sensitive = true and replace their values with "***"
+      def redact_sensitive_inputs
+        @run_data[:profiles]&.each do |p|
+          p[:inputs]&.each do |i|
+            next unless i[:options][:sensitive]
+
+            i[:options][:value] = "***"
+          end
+        end
+      end
+
+      # Optionally suppress diff output in the message field
+      def suppress_diff_output
+        return if @config[:runtime_config][:diff]
+
+        @run_data[:profiles]&.each do |p|
+          p[:controls]&.each do |c|
+            c[:results]&.each do |r|
+              next unless r[:message] # :message only set on failure
+
+              pos = r[:message].index("\n\nDiff:")
+              next unless pos # Only textual tests get Diffs
+
+              r[:message] = r[:message].slice(0, pos)
+            end
+          end
+        end
+      end
+
+      # Optionally sort controls within each profile in report
+      def sort_controls
+        sort_type = @config[:runtime_config][:sort_results_by]
+        return if sort_type == "none"
+
+        @run_data[:profiles]&.each do |p|
+          p[:controls] ||= []
+          p[:groups] ||= []
+
+          case sort_type
+          when "control"
+            p[:controls].sort_by! { |c| c[:id] }
+          when "random"
+            p[:controls].shuffle!
+          when "file"
+            # Sort the controls by file, but preserve order within the file.
+            # Files are called "groups" in the run_data, and the filename is in the id.
+            sorted_control_ids = p[:groups].sort_by { |g| g[:id] }.map { |g| g[:controls] }.flatten
```

Comment:
I'd like to test this out (will assign PR to myself to do so) because I think what some have been asking for is InSpec to give results in the order that they wrote their tests, without jumbling them. 
If that's what this option does, then users might expect sort `none` to _not_ jumble results in the way it currently does. 

Sort by `control` won't work for benchmarks like CIS because they are ordered by section number but those come out wrong when sorted alphanumerically

```
 %w{cis_1.10_ssh_options cis_1.9_firewall_settings cis_1.8_login_settings}.sort
=> ["cis_1.10_ssh_options", "cis_1.8_login_settings", "cis_1.9_firewall_settings"]
```

Reply:
Beg pardon, InSpec does NOT currently jumble results inside one file. I think the case that bugs customers is when they have multiple files (commonly one per control) - and that's exactly what you're fixing here.

Would it be possible to make `file` the default, so users get this desired behaviour for free? 

This does still have the problem mentioned in my last comment - control `1.11` from `cis-1.11.rb`will appear before control `1.2` from `cis-1.2.rb` - I don't think that should be fixed in this PR. Unless maybe to document that alphanumeric ordering isn't magic and InSpec can only put 1.11 before 1.2.